### ISSUE TITLE
ImageReference should support tag and digest

### ISSUE THREAD

**Issue by briandealwis** [GoogleContainerTools/jib#1481]
According to [the Docker spec](https://github.com/docker/distribution/blob/0d3efadf0154c2b8a4e7b6621fff9809655cc580/reference/reference.go#L6
), it's ok to have a reference with both a tag *and* a digest such as `ubuntu:latest@sha256:868fd30a0e47b8d8ac485df174795b5e2fe8a6c8f056cc707b232d65b8a1ab68`

See also: google/go-containerregistry#351

The following test for `ImageReferenceTest` fails
```java
  @Test
  public void testParse_tagAndDigest() throws InvalidImageReferenceException {
    String imageReferenceString =
        "ubuntu:latest@sha256:868fd30a0e47b8d8ac485df174795b5e2fe8a6c8f056cc707b232d65b8a1ab68";
    ImageReference imageReference = ImageReference.parse(imageReferenceString);

    Assert.assertEquals("registry.hub.docker.com", imageReference.getRegistry());
    Assert.assertEquals("ubuntu", imageReference.getRepository());
    Assert.assertEquals("latest", imageReference.getTag());
    Assert.assertEquals(
        "sha256:868fd30a0e47b8d8ac485df174795b5e2fe8a6c8f056cc707b232d65b8a1ab68",
        imageReference.getTag());
    Assert.assertTrue(imageReference.isTagDigest());
  }
```

**Comment by TadCordle** [comment]
@MaartenMoens @jonahgeorge @jhuit We've released v2.2.0, which support image references with both a tag and digest.

**Comment by chanseokoh** [comment]
Note when using both a tag and a digest (for a base image), the tag is purely for documentation. The digest will determine the image (manifest/manifest list).

For the target image, I think it doesn't make sense to put a digest; you shouldn't do it.
